### ISSUE TITLE
Can't export video

### ISSUE THREAD

**Issue by AustinGil** [astrofox-io/astrofox#27]
Hey, this seems like a really awesome project. I got everything working fine and was able to create a project and import the audio clip, but for some reason, when I try to export the video, it just doesn't work. 

The progress bar goes all the way to 100% and then it says it's trying to output the file, but nothing happens.

Note that I am using the Linux version and my audio file is over an hour long, so it takes a very long time. Could it have something to do with this?

**Comment by mikecao** [comment]
Can you trying rendering a shorter clip, like 10 seconds, and see if that finishes? You can also watch your system for the ffmpeg process and see if it's still running.

**Comment by AustinGil** [comment]
Yeah, I thought about this already and tried with a 7 second audio and it still did not work.

**Comment by mikecao** [comment]
I'll see what I can find, but I don't have a native Linux install, only a VM and some things are buggy.

**Comment by AustinGil** [comment]
Yeah, I understand. Most folks dont use Linux so Im never surprised if something doesnt work. I noticed the project is built on Electron. Is it possible for it to just work in the browser then?

**Comment by mikecao** [comment]
The core of the application could run in a browser, but none of the file saving or rendering because that relies on Node.

So I did some testing in my VM and it looks like the rendered frames aren't being piped to the ffmpeg process. I don't really have a solution since it works in Windows and Mac, but at least I know what the problem is.

**Comment by AustinGil** [comment]
Oh yeah, that makes sense. I've run into some other ffmpeg issues before. Im not sure if it's installed with Ubuntu by default, but I do have it installed now, so Im not sure what the problem is. Either way, it doesn't sound like something that is easily solved. The project is still really cool, and I hope I can use it some day :)

**Comment by mikecao** [comment]
@AustinGil Can you try running the latest version from master? I've made some updates the video rendering. See if it works for you. You should be able to open the developer tools and see messages from ffmpeg.

**Comment by AustinGil** [comment]
I downloaded the master branch an unpacked it, then went through the install scripts. I tried running `yarn start` and got 

```
Error launching app
Unable to find Electron app at /home/austin/Desktop/astrofox-master/app

Cannot find module '/home/austin/Desktop/astrofox-master/app'
Require stack:
- /home/austin/Desktop/astrofox-master/node_modules/electron/dist/resources/default_app.asar/main.js
- 
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

After that I tried `yarn dev` and got things to run in the terminal, but the browser showed the following errors.

![image](https://user-images.githubusercontent.com/4879075/97593291-c71d5100-19be-11eb-9b30-129861878516.png)

**Comment by mikecao** [comment]
Sorry I missed a step. You need to run `yarn init-app` first. I'll need to update the scripts later.

**Comment by mikecao** [comment]
Made some updates, pull the latest from master.

**Comment by mikecao** [comment]
You missed the `yarn dev` step. I should probably update the README to be more clear about running a dev build vs a normal build.

**Comment by AustinGil** [comment]
Well, to be honest with you, I tried `yarn dev` as well and it did not work, but I dont remember what the issue was. I just assumed the right command was `yarn start`

**Comment by pymaldebaran** [comment]
I have the exact same problem as @AustinGil : running astrofox under Fedora Linux using the AppImage everything works fine until I try to render the video.
I could build tha app from master following the instruction here but got stuck with a black window when launching app via `yarn start`. exact same error as @AustinGil : 
```
yarn run v1.22.5
$ dotenv electron ./app
(electron) 'allowRendererProcessReuse' is deprecated and will be removed.
[7684:1110/104221.296440:ERROR:gl_utils.cc(316)] [.RendererMainThread-0x3c468b5d3700] GL_INVALID_OPERATION: Framebuffer is incomplete.
[7684:1110/104221.296743:ERROR:gl_utils.cc(316)] [.RendererMainThread-0x3c468b5d3700] GL_INVALID_OPERATION: Framebuffer is incomplete.
[7684:1110/104221.299760:ERROR:gl_utils.cc(316)] [.RendererMainThread-0x3c468b5d3700] GL_INVALID_FRAMEBUFFER_OPERATION: Framebuffer is incomplete.
```
Note: the framebuffer message does not show immediatly but only after some few seconds (or if I click on the network tab of the debugger).

**Comment by 4matic** [comment]
I faced exactly the same problem on Linux. Used latest version from master.
Found in developer tools console after many tries to export video:
```
Error: Uncaught Error: spawn ../astrofox/bin/ffmpeg EACCES
```

```
chmod a+x astrofox/bin/ffmpeg
```
So I fixed binary permission with the command above and webm video export is working!
But exporting to mp4 looks like corrupted. After the process appears to be complete, the application turns completely black, no controls, nothing.

**Comment by mikecao** [comment]
Please try the latest version 1.2.0

**Comment by AustinGil** [comment]
Thanks @mikecao. First run seems to be working. Tested on a very short clip. I'll try a normal sized file next. So far so good :)

**Comment by Osiris-Team** [comment]
@mikecao Had the same problem. Couldnt export videos or images. My antivirus(Avast) blocked it somehow.
By disabling it it works...

**Comment by mikecao** [comment]
Astrofox 2.0 replaced the old rendering pipeline with an offline, frame-by-frame ffmpeg export (no more real-time capture on desktop). Export now has a progress counter, a Cancel button, and always kills ffmpeg and restores the editor when it finishes or fails.

Closing as resolved in Astrofox 2.0. If you can still reproduce this on 2.x, please open a new issue with steps and your version.
